### PR TITLE
Correctly propagate errors from Batch DML

### LINKED ISSUE #148: DML queries must check the response object for error codes.
The `ExecuteBatchDml` method returns a response object that has a `Status` field describing success/failure of the operation. Failure need to be parsed and propagated to the end user.

https://cloud.google.com/spanner/docs/reference/rpc/google.spanner.v1#executebatchdmlresponse

### PR DESCRIPTION
**Pull request by dzou**
Make DML queries propagate error from the response object if present.

[ExecuteBatchDmlResponse contains a Status](https://cloud.google.com/spanner/docs/reference/rpc/google.spanner.v1#executebatchdmlresponse) which contains message if DML query fails. 

Fixes #148

### CODE REVIEW

**Comment by dzou** [comment]
Not quite ready just yet; I realized this needs to be tweaked to correctly handle the case where some statements succeed and the last one fails.

File: src/main/java/com/google/cloud/spanner/r2dbc/client/GrpcClient.java [hunk 1]
```diff
@@ -230,6 +231,14 @@ public GrpcClient(GoogleCredentials credentials) {
               }
             });
       }
+    })
+    .handle((response, sink) -> {
+      if (response.hasStatus() && response.getStatus().getCode() != Status.Code.OK.value()) {
+        sink.error(
+            new R2dbcNonTransientResourceException(response.getStatus().getMessage()));
```
Comment (dmitry-s) [inline src/main/java/com/google/cloud/spanner/r2dbc/client/GrpcClient.java]: can we return status code to the user as well?

Comment (dzou) [reply src/main/java/com/google/cloud/spanner/r2dbc/client/GrpcClient.java]: We could but that would require us creating new exception objects for which I think we ought defer to a later PR. Created: #154

File: src/test/java/com/google/cloud/spanner/r2dbc/it/SpannerIT.java [hunk 2]
```diff
@@ -260,6 +260,14 @@ public void testRunDmlAfterTransaction() {
         .verifyComplete();
   }
 
+  @Test
+  public void testDmlExceptions() {
+    StepVerifier.create(
+        Mono.from(connectionFactory.create())
+            .flatMapMany(conn -> conn.createStatement("INSERT BOOKS asdfasdfasdf").execute()))
```
Comment (ChengyuanZhao) [inline src/test/java/com/google/cloud/spanner/r2dbc/it/SpannerIT.java]: love that book.

File: src/main/java/com/google/cloud/spanner/r2dbc/client/GrpcClient.java [hunk 3]
```diff
@@ -200,7 +202,7 @@ public GrpcClient(GoogleCredentials credentials) {
   }
 
   @Override
-  public Mono<ExecuteBatchDmlResponse> executeBatchDml(
+  public Flux<ResultSet> executeBatchDml(
```
Comment (dmitry-s) [inline src/main/java/com/google/cloud/spanner/r2dbc/client/GrpcClient.java]: Why is it Flux now?

Comment (dzou) [reply src/main/java/com/google/cloud/spanner/r2dbc/client/GrpcClient.java]: Had to make it flux in order to append the error to the end of the resultSets flux if the status != OK. This is equivalent to the previous pattern of doing the flatMapIterable immediately after calling `executeBatchDml`.

Comment (dzou) [reply src/main/java/com/google/cloud/spanner/r2dbc/client/GrpcClient.java]: Also, just added the test we discussed: https://github.com/GoogleCloudPlatform/cloud-spanner-r2dbc/pull/151/commits/ca7e157d6c9dcf0708a88900e5632b9ee2c8b085

PTAL.

**Pull request merged**